exercise/cache/content.py

```diff
@@ -79,6 +79,14 @@ def recursion(
                     'allow_assistant_viewing': False,
                     'children': [],
                 }
+                if self.instance.content_numbering in (
+                        CourseInstance.CONTENT_NUMBERING.NONE,
+                        CourseInstance.CONTENT_NUMBERING.HIDDEN,
+                        ):
+                    entry['hierarchical_name'] = f"{entry['number']} {entry['name']}"
+                else:
+                    entry['hierarchical_name'] = f"{entry['number']} {' '.join(entry['name'].split(' ')[1:])}"
```

Comment:
Would it be simpler to do this? `entry['name']` is annoying to use here since it may or may not have the number at the start. `o.name` never has any numbering prefixed.
```python
entry['hierarchical_name'] = f"{entry['number']} {o.name}"
```